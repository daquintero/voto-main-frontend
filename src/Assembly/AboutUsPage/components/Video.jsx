import React from 'react';

const Video = () => (
    <>
      <div className="featured-video">
        <div className="video">
          <iframe
            title="Bienvenido a Voto Informado"
            src="https://www.youtube.com/embed/1cJ2yb3RJw8"
            frameBorder="0"
            allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen
          />
        </div>
      </div>
    </>
);
export default Video;
