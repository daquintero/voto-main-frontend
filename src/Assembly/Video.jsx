import React from 'react';

const Video = () => (
    <>
      <div className="featured-video">
        <div className="video">
          <iframe
            title="Bienvenido a Voto Informado"
            src="https://www.youtube.com/embed/gmwbcbNdN-0"
            frameBorder="0"
            allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen
          />
        </div>
      </div>
    </>
);
export default Video;
