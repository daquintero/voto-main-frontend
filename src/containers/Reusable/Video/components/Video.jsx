// Video Component for Resizable Areas
import React from 'react';
import PropTypes from 'prop-types';

const Video = info => (
  <div className="video">
    <iframe
      title={info.title}
      src={info.embedUrlSrc}
      frameBorder="0"
      allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
      allowFullScreen
    />
  </div>
);

Video.propTypes = {
  info: PropTypes.shape({
    title: PropTypes.string.isRequired,
    embedUrlSrc: PropTypes.string.isRequired,
  }),
};

Video.defaultProps = {
  info: {
    title: '',
    embedUrlSrc: '',
  },
};

export default Video;
