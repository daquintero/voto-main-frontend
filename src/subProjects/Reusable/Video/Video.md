## Video

This video component resizes
to 100% of the parent container.
