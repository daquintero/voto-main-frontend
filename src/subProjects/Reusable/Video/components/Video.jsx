// Video Component for Resizable Areas
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class Video extends PureComponent {
  static propTypes = {
    title: PropTypes.string,
    embedUrlSrc: PropTypes.string,
  };
  static defaultProps = {
    title: '',
    embedUrlSrc: '',
  }
  render() {
    const { title, embedUrlSrc } = this.props;
    return (
      <div className="video">
        <iframe
          title={title}
          src={embedUrlSrc}
          frameBorder="0"
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen
        />
      </div>
    );
  }
}
