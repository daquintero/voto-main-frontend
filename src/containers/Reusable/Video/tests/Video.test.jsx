import React from 'react';
import renderer from 'react-test-renderer';
import Video from '../components/Video';

describe('<Video />', () => {
  it('Video renders correctly without crashing', () => {
    const tree = renderer
      .create(<Video />)
      .toJSON();
    expect(tree)
      .toMatchSnapshot();
  });
});
