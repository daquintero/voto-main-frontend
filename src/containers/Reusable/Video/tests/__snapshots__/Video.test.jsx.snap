// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Video /> Video renders correctly without crashing 1`] = `
<div
  className="video"
>
  <iframe
    allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
    allowFullScreen={true}
    frameBorder="0"
  />
</div>
`;
